tensorflow/python/ops/resource_variable_ops.py

```diff
@@ -2132,12 +2132,6 @@ def op(self):
     return self._parent_op
 
 
-@ops.RegisterGradient("ReadVariableOp")
```

Comment:
Ah, that explains the errors.  Is there any other way to avoid the duplication?

Reply:
There is a current plan to use C++ gradients from Python, but they're over at https://github.com/tensorflow/tensorflow/tree/master/tensorflow/c/experimental/gradients

I don't see resource op gradients there yet. It's probably something we'll want eventually. (CC @saxenasaurabh)

Unfortunately there have been several "write our gradients in C++" efforts over the years, and this set is not compatible with eager execution. So since we're supporting eager execution in Python we won't be able to use these.

Reply:
Oh, this is indented for tf Java, which also supports eager execution, should I move this there (and to that format)?

Reply:
I'm not sure where Java gets its gradient definitions. Probably better to start with a unit test for the behavior you want.

Reply:
It uses the C API.  And yeah, it looks like I'll have to test it from that end as well.